File: rb/features.rb
# CrowdSourcedLyrics SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CrowdSourcedLyricsFeatures
  def self.make_feature(name)
    case name
    when "base"
      CrowdSourcedLyricsBaseFeature.new
    when "test"
      CrowdSourcedLyricsTestFeature.new
    else
      CrowdSourcedLyricsBaseFeature.new
    end
  end
end

File: py/crowdsourcedlyrics_sdk/features.py
# CrowdSourcedLyrics SDK feature factory

from crowdsourcedlyrics_sdk.feature.base_feature import CrowdSourcedLyricsBaseFeature
from crowdsourcedlyrics_sdk.feature.test_feature import CrowdSourcedLyricsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CrowdSourcedLyricsBaseFeature(),
        "test": lambda: CrowdSourcedLyricsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
